localstack/services/awslambda/invocation/docker_runtime_executor.py

```diff
@@ -267,7 +274,10 @@ def start(self, env_vars: dict[str, str]) -> None:
         self.ip = CONTAINER_CLIENT.get_container_ipv4_for_network(
             container_name_or_id=self.id, container_network=network
         )
-        self.executor_endpoint.container_address = self.ip
+        if config.LAMBDA_ASF_DEV_PORT_EXPOSE:
+            self.executor_endpoint.container_address = "localhost"
+        else:
+            self.executor_endpoint.container_address = self.ip
```

Comment:
```suggestion
        if config.LAMBDA_ASF_DEV_PORT_EXPOSE:
            self.ip = "127.0.0.1"
        self.executor_endpoint.container_address = self.ip
```
This would set both the ip and address correctly

Reply:
The change from localhost to 127.0.0.1 is mostly to avoid some possible ipv6 issues, and, more importantly, the parameter is called `ip`.

Reply:
aha, `self.ip` should also point to localhost and not the detected network address

Reply:
Will not really make sense in this case, as we cannot connect to it anyway. It is currently unused, we can reevaluate in the worst case but for now I would say it should be the address where the container is actually reachable at (if even only at the invocation port).